packit_service/worker/reporting/utils.py

```diff
@@ -90,3 +89,41 @@ def update_message_with_configured_failure_comment_message(
         else ""
     )
     return f"{comment}{configured_failure_message}"
+
+
+def has_identical_comment_in_comments(
+    body: str,
+    comments: Iterable,
+    packit_user: str,
+    mode: DuplicateCheckMode = DuplicateCheckMode.check_last_comment,
+) -> bool:
+    """Check if the body matches provided comments based on the duplication mode."""
+    if mode == DuplicateCheckMode.do_not_check:
+        return False
+
+    for comment in comments:
+        if comment.author.startswith(packit_user):
+            if mode == DuplicateCheckMode.check_last_comment:
+                return body == comment.body
+            if mode == DuplicateCheckMode.check_all_comments and body == comment.body:
+                return True
+    return False
+
+
+def comment_without_duplicating(
+    body: str,
+    pr_or_issue: Union[PullRequest, Issue],
+    mode: DuplicateCheckMode = DuplicateCheckMode.check_last_comment,
+):
+    """
+    Comment on a given pull request/issue, considering the duplication mode.
+    """
+    packit_user = ServiceConfig.get_service_config().get_github_account_name()
+    comments = pr_or_issue.get_comments(reverse=True)
```

Comment:
Yup… this feels exactly like one of the possible bottlenecks caused by lists instead of pure iterators 👀  I wonder how much time it would take to go through one of our validation PRs…